client: Implement command timeouts

foxcpp opened the pull request:
Closes #88.

emersion commented on client.go:
```diff
@@ -38,6 +39,12 @@ type Client struct {
+	CommandTimeout time.Duration
+
+	// Time to wait for responses after final dot.
+	DotTimeout time.Duration
```
How does this interact with future `CHUNKING` support?

foxcpp replied:
I believe that could be mapped to the timeout for response to `BDAT ... LAST`. Do you think it should have a better name?

emersion replied:
`SubmissionTimeout` maybe?

kayrus commented:
Any updates on this PR?

foxcpp commented:
> Any updates on this PR?

I have a lot of IRL things to do. Feel free to finish this PR if you're interested.

foxcpp commented:
I renamed DotTimeout into SubmissionTimeout and also rebased PR onto upstream master so tests could run (there is new vet rule in Go 1.15 that failed)

emersion reviewed (approved):
LGTM, thanks!
